### ISSUE TITLE
Disk drives not attaching at startup

### ISSUE THREAD

**Issue by fatboychummy** [SwitchCraftCC/issues#31]
### Describe the bug

Not sure if this is an issue with CC itself when TPS is low, or if this is an SC issue, but I'll report it here as I can't replicate it in a singleplayer world. I'll need to wait until the server isn't so heavily in-use until I can test if TPS is affecting it (at the time of testing, a small program I wrote was outputting 10-13 TPS).

Onto the issue: A disk drive with a `startup.lua` file on it will not be ran at computer startup. It will very infrequently be ran, but about 98% of the time it will not detect the disk drive exists until a tick or so after the computer starts up.

Of 533 total startups, only 5 startups actually booted to the disk. The disk becomes available after a `disk front` event is queued, after the computer starts up.

### To reproduce

1. Place computer and disk drive side-by-side
2. Put a startup file on the disk drive with anything in it
3. Reboot the computer repeatedly
4. The computer will, 98% of the time, not boot into the disk drive's startup program.

### Expected behavior

The computer should boot into the disk drive's startup program on startup.

### Relevant log output

```sh
Not logs, however these are the files I used for testing:

/disk/startup.lua:
https://p.sc3.io/HfTubJTnkT

/startup.lua:
https://p.sc3.io/Bg7e4PYEJu

I also had another computer just running the following:

while true do
  peripheral.call("left", "reboot")
  sleep(3)
end
```

### Please confirm

- [X] This is not an exploit

**Comment by SquidDev** [comment]
Yeah, this is a CC issue. https://github.com/cc-tweaked/CC-Tweaked/commit/edb21f33be4f5847b46ab04df6b7689596c09f54 changed how disks are mounted, meaning we now defer by one tick to avoid some thread-safety issues. Which, depending on timing, may mean the disk isn't present when the computer comes to boot.

**Comment by Lemmmy** [comment]
Do you think we should move this issue to the CC:T repo?

**Comment by hugeblank** [comment]
This and #10 should, probably

**Comment by SquidDev** [comment]
I have opened https://github.com/cc-tweaked/CC-Tweaked/issues/1399 to track the underlying cause.

We believe #10 is the same as https://github.com/cc-tweaked/CC-Tweaked/issues/1146. Though I don't think anybody has been reliably able to reproduce either, so it's quite hard to know for sure.
